dev/reproducibility.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+import json
+import os
+import shutil
+import nibabel as nib
+
+import numpy as np
+import pandas as pd
+
+from ivadomed import main as ivado
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", "--config", required=True, nargs="+", help="Base config file path.")
+    parser.add_argument("-n", "--iterations", default=10, type=int, help="Number of Monte Carlo iterations.")
+    parser.add_argument("-o", "--output-path", nargs="+", dest="output_path", required=True,
+                        type=str, help="Output directory name without extention to save final csv file. There should "
+                                       "be the same number of output files parameters as the number of config files.")
+    return parser
+
+
+def get_results(context):
+    context["command"] = "eval"
+    pred_mask_path = os.path.join(context["log_directory"], "pred_masks")
+    if os.path.exists(pred_mask_path):
+        shutil.rmtree(pred_mask_path)
+
+    # RandomAffine will be applied during testing
+    if "dataset_type" in context["transformation"]["RandomAffine"]:
+        del context["transformation"]["RandomAffine"]["dataset_type"]
+    return ivado.run_command(context)
+
+
+def compute_csa(config, df_results):
+    subject_list = list(df_results.index)
+    df_results = df_results.assign(csa_pred="", csa_gt="", absolute_csa_diff="", relative_csa_diff="")
+    for subject in subject_list:
+        # Get GT csa
+        gt_path = os.path.join(config["loader_parameters"]["bids_path"], "derivatives", "labels", subject.split("_")[0],
+                               "anat", subject + config["loader_parameters"]["target_suffix"][0] + ".nii.gz")
+        os.system(f"sct_process_segmentation  -i {gt_path} -append 1 -perslice 0 -o csa.csv")
+        df = pd.read_csv("csa.csv")
+        csa_gt = df["MEAN(area)"][0]
+        os.system("rm csa.csv")
+
+        # Get prediction csa
+        pred_path = os.path.join(config["log_directory"], "pred_masks", subject + "_pred.nii.gz")
+        pred_nii = nib.load(pred_path)
+        # Keep only first label to compute csa
+        single_label_pred = nib.Nifti1Image(pred_nii.get_fdata()[..., 0], pred_nii.affine)
+        single_label_pred_path = "pred_single_label.nii.gz"
+        nib.save(single_label_pred, single_label_pred_path)
+        os.system(f"sct_process_segmentation  -i {single_label_pred_path} -append 1 -perslice 0 -o csa.csv")
```

Comment:
:+1:

Reply:
for our study we probably want to estimate CSA over very small number of slices (1-5). the more slices, the less benefits we will observe with the soft vs hard seg

Reply:
so you could do perslice 1 or perslice 0 -z 5:7 (for example)

Reply:
Should we go with: median slice (in IS) for each image?

Reply:
> so you could do perslice 1 or perslice 0 -z 5:7 (for example)

Some images (eg ucl) have a very low number of slices (eg 3).
What about measuring on all the slices, then pick the median slice in the csv? or the 3 median slices (median +/- 1)?

Reply:
medial slice sounds good

Reply:
Updated here: 83bde28